app/forms/registration_form.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class RegistrationForm < ApplicationForm
+  def save(create_params)
+    return false unless validate_client(create_params[:client_id])
+
+    user = User.new(create_params.except(:client_id))
+
+    unless user.valid?
+      user_error = user.errors
+      errors.add(:user, user_error.full_messages.last) unless user_error.empty?
```

Comment:
```suggestion
      user_errors = user.errors
      errors.add(:user, user_errors.full_messages.last) unless user_errors.empty?
```

Also, do not feel obliged to solve all the warnings from `reek` (e.g. `calling user.errors twice`). These warnings make sense in case `.errors` was a resource heavy method (e.g. heavy calculations/processing without caching) — but `errors` is just a hash so we can call `user.errors.XXX` without any performance issue — or at least the perf difference is negligeable.